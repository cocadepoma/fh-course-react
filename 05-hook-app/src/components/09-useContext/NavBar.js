import React from "react";
import { NavLink } from "react-router-dom";

export const NavBar = () => {
    return (
        <nav className='navbar navbar-expand-lg navbar-dark bg-dark'>
            <div className='container-fluid'>
                <div className='navbar-nav'>
                    <NavLink
                        className='nav-link'
                        activeClassName='active'
                        exact
                        to='/'
                    >
                        Home
                    </NavLink>
                    <NavLink
                        className='nav-link'
                        activeClassName='active'
                        exact
                        to='/about'
                    >
                        About
                    </NavLink>
                    <NavLink
                        className='nav-link'
                        activeClassName='active'
                        exact
                        to='/login'
                    >
                        Login
                    </NavLink>
                </div>
            </div>
        </nav>
    );
};
