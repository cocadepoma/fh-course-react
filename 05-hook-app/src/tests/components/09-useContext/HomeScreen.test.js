import { mount } from "enzyme";
import React from "react";
import { HomeScreen } from "../../../components/09-useContext/HomeScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe("pruebas en <HomeScreen/>", () => {
    const user = {
        name: "Paco",
        email: "paco@gmail.com",
    };
    // shallow() renderiza el primer componente, con lo que HomeScreen lo mostrar√° <HomeScreen/>
    // para que renderize todo, tenemos que hacerlo con mount
    const wrapper = mount(
        <UserContext.Provider value={{ user }}>
            <HomeScreen />
        </UserContext.Provider>
    );
    test("debe de mostrarse correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });
});
