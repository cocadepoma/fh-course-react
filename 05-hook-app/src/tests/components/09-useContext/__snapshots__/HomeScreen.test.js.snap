// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <HomeScreen/> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Homescreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Paco",
   "email": "paco@gmail.com"
}
  </pre>
</div>
`;
